nixpkgs/nixpkgs.bzl

```diff
@@ -101,3 +103,68 @@ nixpkgs_package = repository_rule(
   local = True,
   environ = ["NIX_PATH"],
 )
+
+def _cc_configure_custom(ctx):
+  overriden_tools = {
+    "gcc": ctx.path(ctx.attr.gcc),
+    "ld": ctx.path(ctx.attr.ld),
+  }
+  return cc_autoconf_impl(ctx, overriden_tools)
+
+cc_configure_custom = repository_rule(
+  implementation = _cc_configure_custom,
+  attrs = {
+    "gcc": attr.label(
+      executable=True,
+      cfg="host",
```

Comment:
This line looks suspicious to me. Why is it needed?

Reply:
If we want `executable` it is obligatory to specify `cfg="host"`, see: https://docs.bazel.build/versions/master/skylark/rules.html#configurations.

> For each label attribute, you can decide whether the dependency should be built in the same configuration, or transition to the host configuration (using cfg). If a label attribute has the flag executable=True, the configuration must be set explicitly.

And later:

> Tools that are executed as part of the build (e.g., compilers, code generators) should be built for the host configuration. In this case, specify cfg="host" in the attribute.

Reply:
OK makes sense.